api/v1beta2/ibmvpccluster_types.go

```diff
@@ -103,25 +114,118 @@ type VPCLoadBalancerStatus struct {
 	ControllerCreated *bool `json:"controllerCreated,omitempty"`
 }
 
+// VPCNetworkSpec defines the desired state of the network resources for the cluster.
+type VPCNetworkSpec struct {
+	// computeSubnetsSpec is a set of Subnet's which define the Compute subnets.
+	ComputeSubnetsSpec []Subnet `json:"computeSubnetsSpec,omitempty"`
+
+	// controlPlaneSubnetsSpec is a set of Subnet's which define the Control Plane subnets.
+	ControlPlaneSubnetsSpec []Subnet `json:"controlPlaneSubentsSpec,omitempty"`
+
+	// resourceGroup is the name of the Resource Group containing all of the newtork resources.
+	// This can be different than the Resource Group containing the remaining cluster resources.
+	ResourceGroup *string `json:"resourceGroup,omitempty"`
+
+	// securityGroups is a set of SecurityGroup's which define the VPC Security Groups that manage traffic within and out of the VPC.
+	SecurityGroups []SecurityGroup `json:"securityGroups,omitempty"`
+
+	// vpc defines the IBM Cloud VPC.
+	VPC *VPCResource `json:"vpc,omitempty"`
+}
+
+// SecurityGroup dummy.
+// TODO(cjschaef): Dummy SecurityGroup until it is defined in a common location.
+type SecurityGroup struct {
+	Name string `json:"name"`
+}
+
+// COSInstance dummy.
+// TODO(cjschaef): Dummy COSInstance until it is defined in a common location.
+type COSInstance struct {
+	Name string `json:"name"`
+}
+
+// VPCResource dummy.
+// TODO(cjschaef): Dummy VPCResource until it is defined in a common location.
+type VPCResource struct {
+	Name string `json:"name"`
+}
```

Comment:
I'll remove these dummy types if the other PR's adding them are merged before.
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1670
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1669